modules/mail.js

```diff
@@ -1,55 +1,97 @@
 'use strict';
+const AWS = require('aws-sdk');
+const config = require('config');
 const debug = require('debug')('blf-alpha:mailer');
+const htmlToText = require('html-to-text');
+const juice = require('juice');
 const nodemailer = require('nodemailer');
-const config = require('config');
 const path = require('path');
-const AWS = require('aws-sdk');
 const Raven = require('raven');
-const juice = require('juice');
-const htmlToText = require('html-to-text');
 
-const renderHtmlEmail = html => {
-    const options = {
-        webResources: {
-            relativeTo: path.resolve(__dirname, '../public')
-        }
-    };
-    return new Promise((resolve, reject) => {
-        juice.juiceResources(html, options, (err, inlinedHtml) => {
-            if (err) {
-                return reject(err);
-            }
-            resolve(inlinedHtml);
-        });
-    });
-};
+const app = require('../server');
 
-// create Nodemailer SES transporter
+/**
+ * Create Nodemailer SES transporter
+ */
 const transport = nodemailer.createTransport({
     SES: new AWS.SES({
         apiVersion: '2010-12-01',
         region: 'eu-west-1'
     })
 });
 
+function renderHtmlEmail(html) {
+    return new Promise((resolve, reject) => {
+        juice.juiceResources(
+            html,
+            {
+                webResources: {
+                    relativeTo: path.resolve(__dirname, '../public')
+                }
+            },
+            (err, inlinedHtml) => {
+                if (err) {
+                    return reject(err);
+                }
+                resolve(inlinedHtml);
+            }
+        );
+    });
+}
+
+/**
+ * genarateHtmlEmail
+ *
+ * Given an email schema generate full email HTML
+ * - Render template through express/template-engine
+ * - Generate full emai HTML and return alongside orignal schema
+ *
+ * @param {Object} emailsToGenerate
+ * e.g. {
+ *   name: 'example,
+ *   templateName: 'emails/someTempalte',
+ *   templateData: { … }
+ *   sendTo: 'example@example.com',
+ *   subject: 'The greatest email ever'
+ * ]
+ */
+function genarateHtmlEmail(emailData) {
+    return new Promise((resolve, reject) => {
+        app.render(emailData.templateName, emailData.templateData, (templateErr, html) => {
+            if (templateErr) {
+                reject(templateErr);
+            }
+
+            renderHtmlEmail(html)
```

Comment:
I know I named this method, but could we call it something a bit clearer like `makeStylesInline` etc? At the mo it's not very obvious what the difference is between `generateHtmlEmail` and `renderHtmlEmail`.

Reply:
Good point